### DISCUSSION TITLE
Can't route by port

### DISCUSSION THREAD

**Discussion by fastaria** [XTLS/Xray-core#3410] [Q&A] [answered]
Hi! I want to choose outbound connection by inbound port.

I have this JSON for a server
```json
{
  "inbounds": [
    {
      "protocol": "shadowsocks",
      "port": "1455,1456",
      "settings": {
        "password": "somepass",
        "method": "2022-blake3-chacha20-poly1305",
        "network": "tcp,udp"
      }
    }
  ],
  "outbounds": [
    {
      "protocol": "socks",
      "tag": "tor",
      "settings": {
        "servers": [
          {
            "address": "127.0.0.1",
            "port": 9050
          }
        ]
      }
    },
    {
      "protocol": "freedom",
      "tag": "direct"
    }
  ],
  "routing": {
    "rules": [
      {
        "port": 1455,
        "outboundTag": "direct"
      },
      {
        "port": 1456,
        "outboundTag": "tor"
      }
    ]
  }
}
```

The idea is that when the traffic goes through 1455 inbound port it is forwarded to `direct` outbound and if it is 1456 port it goes through `tor` outbound. 

And the problem is that the traffic always goes through the first outbound in the array despite of the routing rules. So with this config it always goes through `tor`. 

How can I fix this? Do I have a mistake in the config?

**Comment by mmmray** [comment]
in the current setup, no rule takes effect, so first outbound applies

you are confusing "destination port of the user traffic" with "destination port of the vpn"

if a user browses a website like `https://google.com:1455`, your rule will take effect

instead, it is better to create two inbounds, one per port, and add `"tag": "tor"` to one inbound

**Reply by fastaria** [reply]
Thanks, now I understand .

I've split it into two inbounds and added `tor` tag to one of them as you've sad, but the problem stays the same.
If I understand it correctly, the "tag" inbound field doesn't say what outbound to use, but just assigns a tag name to an inbound.

Here is the new config. Am I doing it right?
```json
{
  "inbounds": [
    {
      "protocol": "shadowsocks",
      "port": "1455",
      "settings": {
        "password": "somepass",
        "method": "2022-blake3-chacha20-poly1305",
        "network": "tcp,udp"
      }
    },
    {
      "protocol": "shadowsocks",
      "port": "1456",
      "tag":"tor",
      "settings": {
        "password": "somepass",
        "method": "2022-blake3-chacha20-poly1305",
        "network": "tcp,udp"
      }
    }
  ],
  "outbounds": [
    {
      "protocol": "socks",
      "tag": "tor",
      "settings": {
        "servers": [
          {
            "address": "127.0.0.1",
            "port": 9050
          }
        ]
      }
    },
    {
      "protocol": "freedom",
      "tag": "direct"
    }
  ]
}
```

**Reply by mmmray** [reply, marked as answer]
you need a route like `{"inboundTag": ["my-tag-on-inbound"], "outboundTag": "my-tag-on-outbound"}`

I believe tags on inbound and outbound live in completely separate namespaces

**Reply by fastaria** [reply]
It works, thanks!
